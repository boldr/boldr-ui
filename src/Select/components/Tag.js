/* @flow */
import React, { Component } from 'react';

export type Props = {
  prefixCls?: string,
  cid?: string,
  value?: any,
  text?: any,
  onFocus?: Function,
};

class Tag extends Component {
  constructor(props: Props) {
    super(props);
    this.deleteTagHandler = this.deleteTagHandler.bind(this);
  }

  props: Props;

  deleteTagHandler(evt) {
    evt.preventDefault();
    evt.stopPropagation();
    this.props.onDelete(this.props.cid);
  }

  render() {
    let { prefixCls, text } = this.props;

    return (
      <span>
        {text
          ? <span className={`${prefixCls}-tag`}>
              {text}
              <i className={`${prefixCls}-delete`} onClick={this.deleteTagHandler} />
            </span>
          : ''}
      </span>
    );
  }
}

export default Tag;
