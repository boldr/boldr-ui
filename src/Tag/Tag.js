// @flow
import React from 'react';
import type { Node } from 'react';
import classNames from 'classnames';
import BoldrComponent from '../util/BoldrComponent';

export type Props = {
  children: string,
  id: string,
  thumb?: Node,
  onRemove: Function,
  removable?: boolean,
  // small or large
  size?: string,
  wrap?: boolean,
  disabled?: boolean,
};

class Tag extends BoldrComponent {
  static defaultProps = {
    onRemove: () => {},
    size: 'small',
    removable: true,
  };
  props: Props;
  render(): Node {
    const { id, children, thumb, removable, onRemove, size, wrap, disabled } = this.props;

    const className = classNames({
      'boldr-tag': true,
      'boldr-tag__lg': size === 'large',
      'boldr-tag__wrap': wrap,
      'boldr-tag__disabled': disabled,
    });

    const innerClassName = classNames({
      'boldr-tag__inner-wrap': wrap,
      'boldr-typo-4': true,
    });

    const title = wrap ? children : '';

    return (
      <span className={className} disabled={disabled} id={id} title={title}>
        {thumb && <span className="boldr-tag__thumb">{thumb}</span>}
        <span className={innerClassName}>{children}</span>
        {removable &&
        !disabled && (
          // eslint-disable-next-line
          <a
            className="boldr-tag__remove--btn"
            onClick={() => onRemove(id)}
            onKeyPress={() => onRemove(id)}
          />
        )}
      </span>
    );
  }
}

export default Tag;
