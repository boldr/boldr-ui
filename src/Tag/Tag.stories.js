import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import Icon from '../Icons/Icon';
import Tag from './Tag';

storiesOf('Tag', module)
  .addDecorator(centered)
  .add('Default', () => (
    <Tag id="tag1" removable={false}>
      Your Tag
    </Tag>
  ))
  .add('Removable', () => <Tag id="tag2">Tag removable</Tag>)
  .add('Large', () => (
    <Tag id="tag3" size="large">
      LARGE!
    </Tag>
  ))
  .add('Thumbnail', () => (
    <Tag id="tag4" size="large" thumb={<Icon kind="account-card" color="#333" />}>
      Large with Thumb
    </Tag>
  ));
