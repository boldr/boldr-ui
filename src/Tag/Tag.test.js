/* eslint-env jest */
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import Tag from './Tag';

test('<Tag /> matches the snapshot', () => {
  const wrapper = shallow(<Tag id="a">abc</Tag>);

  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
test('should add the passed "className" prop to the rendered node if passed.', () => {
  const wrapper = shallow(<Tag id="a">abc</Tag>);

  expect(wrapper.is('.boldr-tag')).toBe(true);
});
test('should adapt to size prop', () => {
  const wrapper = shallow(
    <Tag id="a" size="large">
      abc
    </Tag>,
  );

  expect(wrapper.is('.boldr-tag__lg')).toBe(true);
});
