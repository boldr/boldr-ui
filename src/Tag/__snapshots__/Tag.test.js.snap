// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> matches the snapshot 1`] = `
<span
  className="boldr-tag"
  id="a"
  title=""
>
  <span
    className="boldr-typo-4"
  >
    abc
  </span>
  <a
    className="boldr-tag__remove--btn"
    onClick={[Function]}
    onKeyPress={[Function]}
  />
</span>
`;
